src/azure-cli/azure/cli/command_modules/ams/_params.py

```diff
@@ -81,6 +86,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('ams account storage sync-storage-keys') as c:
         c.argument('id', required=True)
 
+    with self.argument_context('ams account storage set-authentication') as c:
+        c.argument('storage_auth', help='Allowed values: {}'.format(", ".join(get_storage_authentication_allowed_values_list())))
```

Comment:
Please levarage `arg_type=get_enum_type(get_storage_authentication_allowed_values_list())` for the allowed value list so it can accept case-insentive values and throw an error automatically when the provided value is not in the value list.
Help message can be a simple description of `storage_auth` argument, `get_enum_type` will automatically append the message of allowed values.